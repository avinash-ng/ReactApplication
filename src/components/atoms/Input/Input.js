import React from 'react'
import PropTypes from 'prop-types'

const Input = (props) => {
    // console.log(props.type + " " + props.value + " " + props.name)
    return ( <
        input type={props.type}
        className={props.className}
        onChange = { props.onChange }
        value = { props.value }
        name = { props.name }
        placeholder = { props.placeholder }
        />
    )
}

Input.propTypes = {
    type: PropTypes.string,
    onChange: PropTypes.func,
    value: PropTypes.string,
    name: PropTypes.string

}
export default Input