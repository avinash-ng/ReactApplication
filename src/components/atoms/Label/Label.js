import React from 'react'
import PropTypes from 'prop-types'
const Label = (props) => {
    return (
        <span className="a__label" >{props.label}</span>
        
    )
}

Label.propTypes = {
    label : PropTypes.string.isRequired
}
export default Label