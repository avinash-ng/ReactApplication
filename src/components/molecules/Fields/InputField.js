import React from 'react'
import Label from "../../atoms/Label/Label"
import Input from "../../atoms/Input/Input"
const InputField = (props) => {
    return (
        <div>
            <Label label={props.label}> </Label>
            <Input className={props.className} type={props.type} name={props.name} value={props.value} onChange={props.onChange} placeholder={props.placeholder}></Input>
        </div>   
    )
}
export default InputField;