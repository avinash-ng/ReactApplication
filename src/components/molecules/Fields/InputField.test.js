import InputField from './InputField'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure, mount } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer'
import { Button } from '@material-ui/core'
import { render } from '@testing-library/react';
import Input from '../../atoms/Input/Input'
import Label from '../../atoms/Label/Label'

configure({ adapter: new Adapter() });
describe('<InputField /> ', () => {
    it('renders <Label /> element properly', () => {
        const RenderComponent = shallow(
            <InputField label="First Name "></InputField>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.find(Label).exists()).toBe(true)
    })
    
    it('renders <Input /> element properly', () => {
        const RenderComponent = shallow(
            <InputField label="Last Name "></InputField>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.find(Input).exists()).toBe(true)
    })
    it('recieved props correctly and check on click functionality', () => {
        const label = 'First Name ';
        const type = 'text';
        const name = 'fname';
        const onChange = sinon.spy();
        const value = "avinash";
        const RenderComponent = shallow(<InputField label={label} name={name} value={value} type={type} onChange={onChange}></InputField>)
        expect(RenderComponent.find(Input).prop('name')).toBe(name)
        
    })


    it('Matches snapshot', () => {
        const label = 'Email ';
        const type = 'email';
        const name = 'email';
        const onChange = sinon.spy();
        const value = "abcd@gmail.com";
        const RenderComponent = mount(<InputField label={label} name={name} value={value} type={type} onChange={onChange}></InputField>)
        expect(RenderComponent).toMatchSnapshot()
    })
})