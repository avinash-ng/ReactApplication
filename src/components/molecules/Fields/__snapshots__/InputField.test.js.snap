// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField />  Matches snapshot 1`] = `ReactWrapper {}`;
